src/main/kotlin/com/ampnet/identityservice/controller/AuthorizationController.kt

```diff
@@ -0,0 +1,47 @@
+package com.ampnet.identityservice.controller
+
+import com.ampnet.identityservice.controller.pojo.request.AuthorizationRequest
+import com.ampnet.identityservice.controller.pojo.request.PayloadRequest
+import com.ampnet.identityservice.controller.pojo.request.RefreshTokenRequest
+import com.ampnet.identityservice.controller.pojo.response.AccessRefreshTokenResponse
+import com.ampnet.identityservice.controller.pojo.response.PayloadResponse
+import com.ampnet.identityservice.service.TokenService
+import com.ampnet.identityservice.service.VerificationService
+import mu.KLogging
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class AuthorizationController(
+    private val verificationService: VerificationService,
+    private val tokenService: TokenService
+) {
+
+    companion object : KLogging()
+
+    @PostMapping("/authorize")
+    fun getPayload(@RequestBody request: PayloadRequest): ResponseEntity<PayloadResponse> {
+        logger.debug { "Received request to get payload for address: ${request.address}" }
+        val payload = verificationService.generatePayload(request.address)
+        return ResponseEntity.ok(PayloadResponse(payload))
+    }
+
+    @PostMapping("/authorize/jwt")
+    fun authorizeJwt(@RequestBody request: AuthorizationRequest): ResponseEntity<AccessRefreshTokenResponse> {
+        logger.debug { "Received request for token with address: ${request.address}" }
+        val payloadValid = verificationService.verifyPayload(request.address, request.signedPayload)
+        if (payloadValid.not()) return ResponseEntity.badRequest().build()
```

Comment:
I would use curly braces with this `if` here.

Reply:
Why? One liners with if don't require curly braces, we mostly use it without the `{}`

Reply:
No special reason, I just think that it is more readable if you do it like that since it has a `return` in it. It's more of a suggestion than a request really.

Reply:
If there was an else clause it would definitely be more readable, to me its more elegant without the `{}`